### ISSUE TITLE
Impement tests for add/remove/get Topic Filters

### ISSUE THREAD

**Issue by ramya-rao-a** [Azure/azure-service-bus-node#153]
**Topic Filters: Get Rules**

- [x] Subscription with 0/1/multiple rules returns rules as expected
- [x] Default rule is returned for the subscription for which no rules were added
- [x] Rule with SQL filter returns expected filter expression
- [ ] Rule with SQL filter and action returns expected filter and action expression
- [x] Rule with Correlation filter returns expected filter
- [ ] Rule with Correlation filter and action (Is this allowed?) returns expected filter and action expression

**Topic Filters: Add Rule**

Positive Test Cases 
- [ ] Add Rule with Boolean filter 
- [x] Add Rule with SQL filter 
- [ ] Add Rule with SQL filter and action 
- [x] Add Rule with Correlation filter 
- [ ] Add Rule with Correlation filter and action (Is this allowed?) 

  
Negative Test Cases 
- [x] Adding a rule with a name which matches with existing rule 
- [x] Adding a rule with no name  
- [x] Adding a rule with no filter 
- [ ] Adding a rule with a filter that is neither boolean, string or a co-relation object
- [ ] Adding a rule with a SQL action whose input is not a string

**Topic Filters: Remove Rule**
- [x] Removing non existing rule on a subscription that doesnt have any rules should throw error
- [x] Removing non existing rule on a subscription that has other rules should throw error
- [x] Removing existing rule on a subscription

**Comment by HarshaNalluru** [comment]
Whenever the rules are added or we see the existing rules, do we have to verify if the rule is working by sending and receiving the data ?
@ramya-rao-a

**Comment by ramya-rao-a** [comment]
I wrote these test cases only for adding/removing/getting of rules, so we dont need to test if the rule actually works

But you are right we should have something to test if the rule actually works :)

Lets use this issue to track just the get/add/remove.
I'll log a separate issue for testing the rules themselves

**Comment by ramya-rao-a** [comment]
The leftover test cases are being blocked by bugs #186 & #187
